### PR TITLE
Validate namespace name before creating it

### LINKED ISSUE #1452: Better error if namespace cannot be created
**Describe the bug**

I am hitting my namespace quota, and I am getting this error:

```
$ okteto create namespace
x failed to create namespace
```

**Expected behavior**
A more descriptive error

### PR DESCRIPTION
**Pull request by jLopezbarb**
Fixes #1452

## Proposed changes
-  Checks if namespace length is larger than the accepted
-  Check if the namespace is malformed (have unsupported characters such as uppercase)

### CODE REVIEW

File: pkg/okteto/namespace.go [hunk 1]
```diff
@@ -110,3 +121,20 @@ func DeleteNamespace(ctx context.Context, namespace string) error {
 	var body DeleteBody
 	return query(ctx, q, &body)
 }
+
+func validateNamespace(namespace string) error {
+	if len(namespace) > MAX_ALLOWED_CHARS {
+		return errors.UserError{
+			E:    fmt.Errorf("Failed to create namespace '%s': Exceeded number of character.", namespace),
```
Comment (rberrelleza) [inline pkg/okteto/namespace.go:128]:
```suggestion
			E:    fmt.Errorf("Invalid namespace name" ),
```

File: pkg/okteto/namespace.go [hunk 2]
```diff
@@ -110,3 +121,20 @@ func DeleteNamespace(ctx context.Context, namespace string) error {
...
+			Hint: "Please try to shorten namespace name.",
+		}
+	}
+	nameValidationRegex := regexp.MustCompile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+	if !nameValidationRegex.MatchString(namespace) {
+		return errors.UserError{
+			E:    fmt.Errorf("Failed to create namespace '%s': Malformed namespace name.", namespace),
```
Comment (rberrelleza) [inline pkg/okteto/namespace.go:135]:
```suggestion
			E:     fmt.Errorf("Invalid namespace name"),
```

File: pkg/okteto/namespace.go [hunk 3]
```diff
@@ -110,3 +121,20 @@ func DeleteNamespace(ctx context.Context, namespace string) error {
 	var body DeleteBody
 	return query(ctx, q, &body)
 }
+
+func validateNamespace(namespace string) error {
+	if len(namespace) > MAX_ALLOWED_CHARS {
+		return errors.UserError{
+			E:    fmt.Errorf("Failed to create namespace '%s': Exceeded number of character.", namespace),
+			Hint: "Please try to shorten namespace name.",
```
Comment (rberrelleza) [inline pkg/okteto/namespace.go:129]:
```suggestion
			Hint: "Namespace must be shorter than 63 characters.",
```

File: pkg/okteto/namespace.go [hunk 4]
```diff
@@ -110,3 +121,20 @@ func DeleteNamespace(ctx context.Context, namespace string) error {
...
+		}
+	}
+	nameValidationRegex := regexp.MustCompile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+	if !nameValidationRegex.MatchString(namespace) {
+		return errors.UserError{
+			E:    fmt.Errorf("Failed to create namespace '%s': Malformed namespace name.", namespace),
+			Hint: "Namespace must start with an alphanumeric character and only supports lowercase and '-'",
```
Comment (rberrelleza) [inline pkg/okteto/namespace.go:136]:
```suggestion
			Hint: "Namespace must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character",
```

File: pkg/okteto/namespace.go [hunk 5]
```diff
@@ -110,3 +121,20 @@ func DeleteNamespace(ctx context.Context, namespace string) error {
 	var body DeleteBody
 	return query(ctx, q, &body)
 }
+
+func validateNamespace(namespace string) error {
+	if len(namespace) > MAX_ALLOWED_CHARS {
```
Comment (rberrelleza) [inline pkg/okteto/namespace.go:126]: could we use a function from client-go instead of doing the validation ourselves?

**Review by rberrelleza — APPROVED** [review]

File: pkg/okteto/namespace.go (hunk 5 above)
Comment (jLopezbarb) [reply pkg/okteto/namespace.go:126]: If we do that we can not get which of the validation failed. The method is validateDNS1123 which checks the length and the regex, but if we want to return which is the one failing we will have to check each one

**Review by rberrelleza — APPROVED** [review]

**Pull request merged**